ch_pipeline/processing/client.py

```diff
@@ -195,35 +185,89 @@ def pending(revision):
     "-f",
     "--fairshare",
     type=float,
-    default=None,
+    default=0.0,
     help="Only submit jobs if the account LevelFS is above this threshold.",
 )
 @click.option(
     "--user-fairshare",
     type=float,
-    default=None,
+    default=0.0,
     help="Only submit jobs if the user LevelFS is above this threshold.",
 )
-def generate(revision, number, max_number, submit, fairshare, user_fairshare):
+@click.option(
+    "-p",
+    "--priority-fairshare",
+    type=float,
+    default=0.0,
+    help="Fairshare threshold for priority jobs. The smaller of `fairshare` or this is used.",
+)
+@click.option("--check-failed", is_flag=True, help="Try to requeue failed jobs.")
+def generate(
+    revision,
+    number,
+    max_number,
+    submit,
+    fairshare,
+    user_fairshare,
+    priority_fairshare,
+    check_failed,
+):
     """Submit pending jobs for REVISION (given as type:revision)."""
+    priority_only = False
 
-    if fairshare or user_fairshare:
-        # TODO: find a better way of supplying the account
-        fs = base.slurm_fairshare("rpp-chime_cpu")
+    fs = base.slurm_fairshare("rpp-chime_cpu")
 
-    if fairshare and fairshare > fs[0]:
+    # If a fairshare limit has been set, check that the current fairshare
+    # is below it.
+    if fairshare > fs[0]:
         click.echo(
-            f"Current fairshare {fs[0]} is lower than threshold {fairshare}. Skipping."
+            f"Current fairshare {fs[0]} is lower than threshold {fairshare}. "
+            "Skipping non-priority."
         )
-        return
+        priority_only = True
 
-    if user_fairshare and user_fairshare > fs[1]:
+    # If a user fairshare limit has been set, check that the current user
+    # fairshare is below it.
+    if user_fairshare > fs[1]:
         click.echo(
             f"Current user fairshare {fs[1]} is lower than threshold {user_fairshare}. "
-            "Skipping."
+            "Skipping non-priority."
+        )
+        priority_only = True
+
+    # If a priority fairshare limit has been set, check if priority jobs
+    # can still be processed
+    if priority_only and priority_fairshare > fs[0]:
+        click.echo(
+            f"Current fairshare {fs[0]} is lower than priority threshold {priority_fairshare}. "
+            "Skipping all."
         )
         return
 
+    if check_failed:
+
+        import shutil
+
+        failed = revision.failed()
+        requeue = {"chimedb_error", "time_limit", "mpi_error"}
+
+        for key, tags in failed.items():
+            if key not in requeue:
+                continue
+            # Delete these tags so they get resubmitted to the queue
+            for tag in tags:
+                path = revision.workdir_path / str(tag)
+                try:
+                    # TODO: any issues with removing the entire directory?
+                    shutil.rmtree(path)
```

Comment:
I'm going back and forth on whether or not to allow this. I'm not a huge fan of allowing this to remove entire directories, but that's also the only way of re-queueing jobs at the moment. Maybe I'll try to figure out a different way to do it